src/plugins/intel_npu/src/plugin/npuw/util.cpp

```diff
@@ -1095,6 +1095,18 @@ void ov::npuw::util::unpack(const ov::SoPtr<ov::ITensor>& from,
 #undef CAST
 }
 
+void transpose_data_f16(const int16_t* src_data, int16_t* dst_data, size_t dim1, size_t dim2, size_t dim3) {
```

Comment:
Do we have already existing function for permute/tranpose?

Naive implementation can be done even simpler by just copying data to the tensor with different strides.
e.g let's assume original layout is NCHW [1, 3, 10, 15] but you need NHWC

1) Create tensor with NHWC layout: ov::Tensor tmp([1, 10, 15, 3], ...);
2) Copy data from existing tensor: orig.copy_to(tmp); <- this basically transpose data as `tmp` has different strides
3) orig.set_shape([1, 10, 15, 3])
4) tmp.copy_to(orig)

Reply:
Tried this implementation particularly:
`    const auto& to_shape = to->get_shape();
    size_t dim1 = to_shape[0];
    size_t dim2 = to_shape[1];
    size_t dim3 = to_shape[2];
    ov::Shape new_shape = {dim2, dim3, dim1};
    ov::Tensor tmp(to->get_element_type(), new_shape);
    ov::SoPtr<ov::ITensor> tmp_itensor = ov::get_tensor_impl(tmp);
    to->set_shape(new_shape);
    to->copy_to(tmp_itensor._ptr);
    tmp_itensor->copy_to(to._ptr);
    std::cout << "To new shape: " << to->get_shape() << std::endl;`
    
    But when I verified, by testing if the actual data is transposed or not. It turns out not. So moved back to earlier implementation.